FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_Plaintext/DemoTasks/PlainTextHTTPExample.c

```diff
@@ -0,0 +1,504 @@
+    #error "Please define democonfigHTTP_PORT in demo_config.h."
+#endif
+
+/* Check that a path for HTTP Method GET is defined. */
+#ifndef democonfigGET_PATH
+    #error "Please define democonfigGET_PATH in demo_config.h."
+#endif
+
+/* Check that a path for HTTP Method HEAD is defined. */
+#ifndef democonfigHEAD_PATH
+    #error "Please define democonfigHEAD_PATH in demo_config.h."
+#endif
+
+/* Check that a path for HTTP Method PUT is defined. */
+#ifndef democonfigPUT_PATH
+    #error "Please define democonfigPUT_PATH in demo_config.h."
+#endif
+
+/* Check that a path for HTTP Method POST is defined. */
+#ifndef democonfigPOST_PATH
+    #error "Please define democonfigPOST_PATH in demo_config.h."
+#endif
+
+/* Check that a request body to send for the POST request is defined. */
+#ifndef democonfigREQUEST_BODY
+    #error "Please define a democonfigREQUEST_BODY in demo_config.h."
+#endif
+
+/* Check that a transport timeout for transport send and receive is defined. */
+#ifndef democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS
+    #define democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS    ( 1000 )
+#endif
+
+/* Check that a size for the user buffer is defined. */
+#ifndef democonfigUSER_BUFFER_LENGTH
+    #define democonfigUSER_BUFFER_LENGTH    ( 2048 )
+#endif
+
+/**
+ * @brief The length of the server's hostname.
+ */
+#define httpexampleSERVER_HOSTNAME_LENGTH     ( sizeof( democonfigSERVER_HOSTNAME ) - 1 )
+
+/**
+ * @brief The length of the HTTP GET method.
+ */
+#define httpexampleHTTP_METHOD_GET_LENGTH     ( sizeof( HTTP_METHOD_GET ) - 1 )
+
+/**
+ * @brief The length of the HTTP HEAD method.
+ */
+#define httpexampleHTTP_METHOD_HEAD_LENGTH    ( sizeof( HTTP_METHOD_HEAD ) - 1 )
+
+/**
+ * @brief The length of the HTTP PUT method.
+ */
+#define httpexampleHTTP_METHOD_PUT_LENGTH     ( sizeof( HTTP_METHOD_PUT ) - 1 )
+
+/**
+ * @brief The length of the HTTP POST method.
+ */
+#define httpexampleHTTP_METHOD_POST_LENGTH    ( sizeof( HTTP_METHOD_POST ) - 1 )
+
+/**
+ * @brief The length of the HTTP GET path.
+ */
+#define httpexampleGET_PATH_LENGTH            ( sizeof( democonfigGET_PATH ) - 1 )
+
+/**
+ * @brief The length of the HTTP HEAD path.
+ */
+#define httpexampleHEAD_PATH_LENGTH           ( sizeof( democonfigHEAD_PATH ) - 1 )
+
+/**
+ * @brief The length of the HTTP PUT path.
+ */
+#define httpexamplePUT_PATH_LENGTH            ( sizeof( democonfigPUT_PATH ) - 1 )
+
+/**
+ * @brief The length of the HTTP POST path.
+ */
+#define httpexamplePOST_PATH_LENGTH           ( sizeof( democonfigPOST_PATH ) - 1 )
+
+/**
+ * @brief Length of the request body.
+ */
+#define httpexampleREQUEST_BODY_LENGTH        ( sizeof( democonfigREQUEST_BODY ) - 1 )
+
+/**
+ * @brief Number of HTTP paths to request.
+ */
+#define httpexampleNUMBER_HTTP_PATHS          ( 4 )
+
+/**
+ * @brief A pair containing a path string of the URI and its length.
+ */
+typedef struct httpPathStrings
+{
+    const char * pcHttpPath;
+    size_t ulHttpPathLength;
+} httpPathStrings_t;
```

Comment:
Should this and next be capitalized?

Reply:
I think the intention here was to not capitalize to show that it is private to this file.